psi4/driver/schema_wrapper.py

```diff
@@ -197,6 +208,145 @@ def _convert_variables(data, context=None, json=False):
     return ret
 
 
+def _convert_basis(basis):
+    """Converts a Psi4 basis object to a QCElemental basis.
+    """
+    centers = []
+    symbols = []
+
+    # Loop over centers
+    for c in range(basis.molecule().natom()):
+        center_shells = []
+        symbols.append(basis.molecule().symbol(c).title())
+
+        # Loop over shells *on* a center
+        for s in range(basis.nshell_on_center(c)):
+            shell = basis.shell(basis.shell_on_center(c, s))
+            if shell.is_pure():
+                dtype = "spherical"
+            else:
+                dtype = "cartesian"
+
+            # Build the shell
+            coefs = [[shell.coef(x) for x in range(shell.nprimitive)]]
+            exps = [shell.exp(x) for x in range(shell.nprimitive)]
+            qshell = qcel.models.basis.ElectronShell(angular_momentum=[shell.am],
+                                                     harmonic_type=dtype,
+                                                     exponents=exps,
+                                                     coefficients=coefs)
+            center_shells.append(qshell)
+
+        centers.append(qcel.models.basis.BasisCenter(electron_shells=center_shells))
+
+    # Take unique to prevent duplicate data, doesn't matter too much
+    hashes = [hash(json.dumps(centers[x].dict(), sort_keys=True)) for x in range(len(centers))]
+
+    uniques = {k: v for k, v in zip(hashes, centers)}
+    name_map = {}
+    counter = {}
+
+    # Generate reasonable names
+    for symbol, h in zip(symbols, hashes):
+        if h in name_map:
+            continue
+
+        if symbol in counter:
+            counter[symbol] += 1
+        else:
+            counter[symbol] = 1
+
+        name_map[h] = f"{basis.name()}_{symbol}{counter[symbol]}"
```

Comment:
because psi doesn't allow setting basis by atom number, you could use `atom.symbol() + atom.label()` for uniqueness. but this is more general anyways.

Reply:
This would always generate a unique label for each atom though? We want to create only unique labels per unique atom/basis combo.

Reply:
Bad diction. I meant what you have is more general anyways. And your scheme will generate fewer labels than my comment.